hedera-node/src/main/java/com/hedera/services/store/models/Token.java

```diff
@@ -0,0 +1,153 @@
+package com.hedera.services.store.models;
+
+/*-
+ * ‌
+ * Hedera Services Node
+ * ​
+ * Copyright (C) 2018 - 2021 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import com.google.common.base.MoreObjects;
+import com.hedera.services.legacy.core.jproto.JKey;
+import com.hederahashgraph.api.proto.java.ResponseCodeEnum;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+
+import static com.hedera.services.exceptions.ValidationUtils.checkInvariant;
+import static com.hedera.services.exceptions.ValidationUtils.validateTrue;
+import static com.hedera.services.utils.MiscUtils.describe;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.INSUFFICIENT_TOKEN_BALANCE;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.INVALID_TOKEN_BURN_AMOUNT;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.INVALID_TOKEN_MINT_AMOUNT;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.TOKEN_HAS_NO_SUPPLY_KEY;
+
+public class Token {
+	private final Id id;
+
+	private boolean supplyHasChanged;
+
+	private long totalSupply;
+	private JKey kycKey;
+	private JKey freezeKey;
+	private JKey supplyKey;
+	private Account treasury;
+	private Account autoRenewAccount;
+
+	public Token(Id id) {
+		this.id = id;
+	}
+
+	public void burn(TokenRelationship treasuryRel, long amount) {
+		checkInvariant(
```

Comment:
Personally, I prefer a code style where we don't have a blank first line. What do you think?

Reply:
Actually, that isn't quite true either. I sometimes like to move the args to the next line, but not indented this far. Usually only 8 char indent.

Reply:
Reduced whitespace in calls to `validateTrue`.